File: selenium-typescript-e2e-template/pages/basePage.ts
import { WebDriver, By, until, WebElement } from 'selenium-webdriver';

export class BasePage {
  constructor(protected driver: WebDriver) {}

  async navigateTo(url: string): Promise<void> {
    await this.driver.get(url);
  }

  async waitForElement(locator: By, timeout: number = 10000): Promise<WebElement> {
    return await this.driver.wait(until.elementLocated(locator), timeout);
  }
}

File: web/selenium-csharp/Pages/BasePage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;

namespace Pages;

public class BasePage
{
    protected readonly IWebDriver Driver;
    protected readonly WebDriverWait Wait;

    public BasePage(IWebDriver driver)
    {
        Driver = driver;
        Wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10));
    }

    public void NavigateTo(string url)
    {
        Driver.Navigate().GoToUrl(url);
    }

    public IWebElement WaitForElement(By locator)
    {
        return Wait.Until(driver => driver.FindElement(locator));
    }
}